File: backend/src/main/java/net/ugurkartal/backend/services/abstracts/PurchaseService.java
package net.ugurkartal.backend.services.abstracts;

import net.ugurkartal.backend.services.dtos.requests.PurchaseRequest;
import net.ugurkartal.backend.services.dtos.responses.PurchaseCreatedResponse;
import net.ugurkartal.backend.services.dtos.responses.PurchaseGetAllResponse;

import java.util.List;

public interface PurchaseService {
    List<PurchaseGetAllResponse> getAllPurchases();
    List<PurchaseGetAllResponse> getPurchasesByProductId(String productId);
    List<PurchaseGetAllResponse> getPurchasesBySupplierId(String supplierId);
    PurchaseCreatedResponse getPurchaseById(String id);
    PurchaseCreatedResponse addPurchase(PurchaseRequest purchaseRequest);
    PurchaseCreatedResponse updatePurchase(String id, PurchaseRequest purchaseRequest);
    PurchaseCreatedResponse changePurchaseStatus(String id, boolean status);
}

File: frontend/src/services/PurchaseService.ts
import axios from "axios";
import {Purchase} from "../types/Purchase.ts";

export default class PurchaseService {
    getAllPurchases() {
        return axios.get('/api/purchases');
    }

    getPurchaseByProductId(productId: string) {
        return axios.get(`/api/purchases/product/${productId}`);
    }

    getPurchaseById(id: string) {
        return axios.get(`/api/purchases/${id}`);
    }

    addPurchase(purchase: Purchase) {
        return axios.post('/api/purchases', purchase);
    }

    updatePurchase(id: string, purchase: Purchase) {
        return axios.put(`/api/purchases/${id}`, purchase);
    }

    changePurchaseStatus(id: string, status: boolean) {
        return axios.put(`/api/purchases/status/${id}?status=${status}`);
    }
}